code/game/machinery/washing_machine.dm

```diff
@@ -14,306 +13,378 @@
+	visible_message("<span class='danger'>[user] shoves [mob_to_insert] into [src].</span>")
+	return TRUE
+
+/// Handles inserting obj/items into the washing machines, checks machines capacity, does a do_after, and then applys appropriate signals and updates machines state
+/obj/machinery/washing_machine/proc/insert_item_into_tub(mob/user, obj/item/item_to_insert)
+	var/item_content_size = item_to_insert.w_class
+	if(max_tub_capacity < (item_content_size + current_tub_capacity))
+		to_chat(user, "<span class='warning'>You try to insert [item_to_insert] into [src] but [src] is too full for [item_to_insert.p_them()]!</span>")
+		return FALSE
+	if(!do_after_once(user, (1 SECONDS * (item_to_insert.w_class / 3)), target = src, attempt_cancel_message = "You stop inserting [item_to_insert] into [src]."))
+		return FALSE
+	if(!user.drop_item())
+		to_chat(user, "<span class='warning'>[item_to_insert] is stuck to your hand!</span>")
+		return FALSE
+	item_to_insert.forceMove(src)
+	RegisterSignal(item_to_insert, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	LAZYADD(inserted_items, item_to_insert)
+	calculate_tub_capacity()
+	update_washing_state()
+	to_chat(user, "<span class='notice'>You insert [item_to_insert] into [src].</span>")
+	return TRUE
 
-		if(length(contents) < 5)
-			if(state in list(OPEN_EMPTY, OPEN_FULL))
-				user.drop_item()
-				W.forceMove(src)
-				state = OPEN_FULL
-			else
-				to_chat(user, "<span class='warning'>The door is closed!</span>")
-		else
-			to_chat(user, "<span class='warning'>[src] is full!</span>")
-		update_icon(UPDATE_ICON_STATE)
+/*
+	* # check_tub_contents()
+	*
+	* Managing tons of contents inside the object is messy business, you never know when something is deleted, teleported, or moved into your obj
+	* without your machine knowing, this clears bad references for GC/tracking purposes and adds references when something is added randomly to
+	* our contents. This ensures we don't nullspace shit (such as a mob dropping a wrench or something) and spit it all back out afterwards even
+	* if we didn't add it our selves!
+	*
+	* This proc is called via COMSIG_PARENT_QDELETING for objects we put inside through the insert proc
+*/
+/obj/machinery/washing_machine/proc/check_tub_contents()
+	if(LAZYLEN(inserted_items))
+		for(var/obj/item/I in inserted_items)
+			if(QDELETED(I) || I.loc != src)
+				if(!QDELETED(I))
+					UnregisterSignal(I, COMSIG_PARENT_QDELETING)
+				LAZYREMOVE(inserted_items, I)
+	if(LAZYLEN(inserted_mobs))
+		for(var/mob/living/L in inserted_mobs)
+			if(QDELETED(L) || L.loc != src)
+				if(!QDELETED(L))
+					UnregisterSignal(L, COMSIG_PARENT_QDELETING)
+				LAZYREMOVE(inserted_mobs, L)
+	for(var/obj/item/I in contents)
+		if(!LAZYIN(inserted_items, I))
+			LAZYADD(inserted_items, I)
+			RegisterSignal(I, COMSIG_PARENT_QDELETING, PROC_REF(check_tub_contents))
+	for(var/mob/living/L in contents)
+		if(!LAZYIN(inserted_mobs, L))
+			LAZYADD(inserted_mobs, L)
```

Comment:
I thought about doing this, however keep two segregated lists helps prevent unwanted atoms from getting caught up in washing machine stuff. For example, I don't really want to iterate through every `atom/obj` since there's a lot of unwanted types in there and it would be less performative to perform checks to exclude every single one of those types.